mt_sat/mt_sat_wrapper.m

```diff
@@ -221,6 +237,24 @@ function mt_sat_wrapper(mtw_nii,pdw_nii,t1w_nii,mtw_jsn,pdw_jsn,t1w_jsn,varargin
     savejson('',provenance,[pwd filesep 'MTsat.json']);
 end
 
+% JSON file for dataset_description
+addDescription = struct();
+addDescription.Name = 'qMRLab Outputs';
+addDescription.BIDSVersion = '1.5.0';
+addDescription.DatasetType = 'derivative';
+addDescription.GeneratedBy.Name = 'qMRLab';
+addDescription.GeneratedBy.Version = qMRLabVer;
+addDescription.GeneratedBy.Container.Type = 'docker (if Docker{enabled = true})';
+addDescription.GeneratedBy.Container.Tag = 'qmrlab/minimal:2.4.1';
+addDescription.GeneratedBy.Name2 = 'Manual';
+addDescription.GeneratedBy.Description = 'Generated example T1map outputs';
```

Comment:
For now we can change this to `Generated example MTsat outputs`, but moving forward, we can pass such data from `nextflow config`.

Reply:
I'm thinking how to fetch the data from the `nextflow config`.

Reply:
I think you can simply add a config variable. Any config variables can be assigned in the workflow exec command, something like:

```
nextflow run mtsatflow_BIDS.nf --description blah (--root)

Reply:
@agahkarakuzu I don't find here the email you sent me regarding adding the config variable to the execution command `(nextflow run mtsatflow_BIDS.nf --description blah)`. 

If I understand correctly though, the user is going to assign a value (blah) to the variable `description`, and this is going to be passed to the wrapper as a field to fill the `addDescription = struct()`. 
Am I right? 

In this case, I think we will assign (in the exec command) several variables to fill the `dataset_description.json` file. For instance, we will need a value for `addDescription.GeneratedBy.Name = 'qMRLab'` another for `addDescription.GeneratedBy.Container.Type = 'docker (if Docker{enabled = true})'`, etc.

Reply:
@jvelazquez-reyes not all fields in the dataset description are required:

![image](https://user-images.githubusercontent.com/9632322/102841995-ff944680-43d3-11eb-871d-2c2014987dec.png)

Some fields, as you noted, can be populated automatically by few conditional checks. We should do whenever possible. 

For the fields that require human input, we should give users and easy option, as you mentioned below to populate the `addDescription.GeneratedBy.Description`: 

> If I understand correctly though, the user is going to assign a value (blah) to the variable description, and this is going to be passed to the wrapper as a field to fill the  **`addDescription.GeneratedBy.Description`**.

For such fields, we can have default values for the cases where user prefers not to indicate:

For example, the default for   **`addDescription.GeneratedBy.Description`** can be `qMRFlow mt_sat pipeline has been executed."

Reply:
Sorry @agahkarakuzu I still don't have the workflow running properly. I understand the idea of adding a config variable (assign a default value) and pass it to the wrapper. I have been trying the following:

1. In `mtsatflow_BIDS.nf`, I'm assigning `params.description = "qMRFlow mt_sat pipeline has been executed"`
2. Then, I'm passing `'description' = $params.description` to the wrapper.
3. In `mt_sat_wrapper` I'm adding this argument using `addParameter` and then I'm just trying to get the value to populate the JSON file.

But, whether I specified `--description blah` or not in the exec command, I get an error stating an undefined value (blah or default) of `description`. I think an intermediate step between 1 and 2 is missing. 

To fetch docker information, I'm trying a similar approach adding a conditional like:
```
if (docker.enabled == true) {
    params.container = "docker"
    params.tag = getTagValue
}
```

Reply:
@jvelazquez-reyes are you trying these locally? Because I cannot see an optional parameter added to the parser of the wrapper here. How do you perform 2? I think that's the key point. 

You may need to use `echo` to pass `$params.description` to the matlab function call. I can be of more help if I could see the code.

Reply:
@agahkarakuzu I finally managed to print `$params.description` on the JSON file. I was trying all these tests locally, but the wrapper was not the problem. Instead, it was the way I was passing the argument to the function call.

Reply:
That's awesome! Great job @jvelazquez-reyes 🎊